File: backend/internal/domain/entities/lesson.go
package entities

import (
	"time"

	"github.com/google/uuid"
)

type LessonType string

const (
	LessonVideo LessonType = "video"
	LessonText  LessonType = "text"
	LessonQuiz  LessonType = "quiz"
	LessonPDF   LessonType = "pdf"
	LessonLink  LessonType = "link"
)

type Lesson struct {
	ID        uuid.UUID
	ModuleID  uuid.UUID
	OrgID     uuid.UUID
	Title     string
	Content   string
	Type      LessonType
	VideoURL  string
	LinkURL   string
	FileURL   string
	Position  int
	CreatedAt time.Time
	UpdatedAt time.Time
}

File: frontend/src/types/lesson.ts
export type LessonType = 'video' | 'text' | 'quiz' | 'pdf' | 'link'

export interface LessonDTO {
  id: string
  module_id: string
  title: string
  type: LessonType
  content: string
  video_url?: string
  link_url?: string
  file_url?: string
  order: number
  created_at: string
  updated_at: string
}

export interface CreateLessonInput {
  title: string
  type: LessonType
  content: string
  video_url?: string
  link_url?: string
  file_url?: string
  order?: number
}

export interface UpdateLessonInput {
  title?: string
  type?: LessonType
  content?: string
  video_url?: string
  link_url?: string
  file_url?: string
  order?: number
}
